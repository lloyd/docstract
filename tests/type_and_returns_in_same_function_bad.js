/** @@type is an alias for @@return when it occurs in a function
 *  block.*/

/**
 * @function foo
 * But having both @@type and @@return is bogus.
 * @returns {string} This is the first return type.
 * @type {boolean}
 */
