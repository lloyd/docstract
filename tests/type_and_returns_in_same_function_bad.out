{
  "args": [
    "Return type redefined (@type and @returns in same function block?)"
  ], 
  "exception_type": "<type 'exceptions.RuntimeError'>", 
  "stack": [
    [
      "<module>", 
      "got = ds.extractFromFile(os.path.join(testDir, test + \".js\"))"
    ], 
    [
      "extractFromFile", 
      "data = self.extract(contents)"
    ], 
    [
      "extract", 
      "self._analyzeBlock(block, context, firstBlock, data)"
    ], 
    [
      "_analyzeBlock", 
      "handler.attach(v, doc, parseData['blockHandler'].tagName)"
    ], 
    [
      "attach", 
      "DocStract.ReturnTagHandler.attach(self, obj, current, blockType)"
    ], 
    [
      "attach", 
      "\"same function block?)\")"
    ]
  ]
}
